src/io/jesi/backpack/macros.cljc

```diff
@@ -108,14 +108,35 @@
               [form]))]
     (cons 'try (mapcat transform body))))
 
+;FIXME rename
+(defn shorthand* [keyfn symbols]
+  (->> symbols
+       (map (fn [sym]
+              [(keyfn sym) sym]))
+       (into {})))
+
 (defmacro shorthand
   "Returns a map with the keywords from the symbol names"
   [& symbols]
-  (into (array-map)
-    (map
-      (fn [sym]
-        [(keyword (name sym)) sym])
-      symbols)))
+  (shorthand* (comp keyword name) symbols))
+
+(defmacro shorthand-str
+  "String `shorthand`. Returns a map with string keys from the symbol names"
+  [& symbols]
+  (shorthand* name symbols))
+
+(defmacro shorthand-assoc
+  ([map sym]
+   `(assoc ~map ~(keyword (name sym)) ~sym))
+  ([map sym & more]
+   `(assoc ~map ~@(->> (cons sym more)
+                       ;;TODO make point-free
+                       ;;(map (juxt (comp keyword name) identity))
+                       ;;(apply concat)
```

Comment:
Yeah not sure why that didn't work... any ideas?

Reply:
Not really. Reduce is fine though